### DISCUSSION TITLE
Text Selection with keyboard

### DISCUSSION THREAD

**Discussion by resistorsoftware** [kovidgoyal/kitty#8166] [General]
Just a quick question! Is it still a thing where there is no chance Kitty will support moving the cursor and selecting text?

I am finding that to be the only thing really missing and while not a deal breaker, I am curious if it will ever be a thing?

**Comment by kovidgoyal** [comment]
There are many, many ways you can achieve this in kitty.
https://sw.kovidgoyal.net/kitty/overview/#the-scrollback-buffer

So dont know what you mean by its not a thing. In fact you can do this
much better in kitty than most other terminal emulators, since you can
use your own comfortable editor to do it.

**Comment by resistorsoftware** [comment]
What I meant is this. 
if my prompt is `$ some fun thing I typed`
I can *option* + *leftarrow* twice to get the cursor to before the letter I, but now I want to press *shift* + *leftarrow* three times and have the words `some fun thing` available to copy so I can paste them somewhere else. 

That has nothing from scrollback buffer or history, it is just brain dead muscle memory for me from editing ... so that was something I could not see being done in Kitty, at least not with the searching I did. But thanks for letting me know it is easy! I will hold out hope!

**Comment by kovidgoyal** [comment]
That has nothing to do with kitty, what you are describing is a shell
prompt. You need to get whatever shell you are using to support whatever
editing modality you prefer. As for doing it with kitty, you can
actually edit shell prompts in editors already, in any terminal, not
just kitty.

**Reply by resistorsoftware** [reply]
I see. So because I am using ZSH, I need to teach ZSH to handle the command line text. OK. Thanks. Interesting and this disconnect in my brain comes from the fact I fired up Ghostty for the first time, and it loaded my ZSH and it did not do this task either, but when I fire up Warp, it too uses my ZSH but instead, it handles this option of just Shift + arrow key highlighting like a champ. So I incorrectly assumed this was a terminal function and not a shell one.

**Comment by kovidgoyal** [comment]
Warp completely overrides shell prompt handling, it runs the shell in a
background pty and does actual prompt editing itself. This is a really
broken approach, but let me not start on a rant about it.

**Reply by resistorsoftware** [reply]
Ok. I looked into this for ZSH and nothing easy or good jumped out there either. the one plugin I tried does not work without a lot of work and it borks the shift+ctrl lifeline born into kitty here, so that means I am stuck with a ZSH that needs some other fix to work. 

Thanks for all your valuable input!
